opds_import.py

```diff
@@ -288,53 +360,148 @@ def import_from_metadata(
             edition, self.metadata_client, replace=policy
         )
 
+        return edition
+
+
+    def import_pool_from_circulationdata(
+            self, circulationdata, associated_metadata, cutoff_date
+    ):
+        """ For the passed-in CirculationData object, see if can find or create a license pool, 
+        then see if can find one or more editions pre-created for that license pool, 
+        then see if can create a work, and also update the editions 
+        with the new license pool and work.  
+        """
+
+        # Locate or create a LicensePool for this book.
+        license_pool, is_new_license_pool = circulationdata.license_pool(self._db)
+        
+        if (cutoff_date 
+            and not is_new_license_pool 
+            and associated_metadata 
+            and associated_metadata.provider_entry_updated < cutoff_date):
+            # We've already imported this book, we've been told
+            # not to bother with books that appeared before a
+            # certain date, and this book did in fact appear
+            # before that date. There's no reason to do anything.
+            return None
+
+        policy = ReplacementPolicy(
+            subjects=True,
+            links=True,
+            contributions=True,
+            rights=True,
+            link_content=True,
+            even_if_not_apparently_updated=True,
+            mirror=self.mirror,
+            http_get=self.http_get,
+        )
+        circulationdata.apply(
+            license_pool, replace=policy
+        )
+
         if license_pool is None:
             # Without a LicensePool, we can't create a Work.
-            self.log.warn(
-                "No LicensePool present for Edition %r, not attempting to create Work.",
-                edition
-            )
-        else:
-            license_pool.presentation_edition = edition
-            work, is_new_work = license_pool.calculate_work(
-                known_edition=edition,
-                even_if_no_author=even_if_no_author,
-            )
-            if work:
-                if immediately_presentation_ready:
-                    # We want this book to be presentation-ready
-                    # immediately upon import. As long as no crucial
-                    # information is missing (like language or title),
-                    # this will do it.
-                    work.set_presentation_ready_based_on_content()
+            error_message = "No LicensePool created for CirculationData %r, will not attempt to continue to create Work." % circulationdata
+            self.log.warn(error_message)
+            raise DBImportException(error_message)
+
+        return license_pool
 
-        return edition
 
-    def extract_metadata(self, feed):
-        """Turn an OPDS feed into a list of Metadata objects and a list of
-        messages.
+    def import_work_from_circulationdata(
```

Comment:
this doesn't take circulation data as an argument, it's more of a cleanup method after the license pool is imported.